frontend/src/v5/ui/routes/dashboard/projects/federations/createFederationForm/createFederationForm.component.tsx

```diff
@@ -0,0 +1,165 @@
+/**
+ *  Copyright (C) 2022 3D Repo Ltd
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { useEffect, useState } from 'react';
+import { useParams } from 'react-router-dom';
+import { formatMessage } from '@/v5/services/intl';
+import { FederationCreationSchema } from '@/v5/validation/federations';
+import { FormSelect } from '@controls/formSelect/formSelect.component';
+import { FormTextField } from '@controls/formTextField/formTextField.component';
+import { FormModal } from '@controls/modal/formModal/formDialog.component';
+import { yupResolver } from '@hookform/resolvers/yup';
+import { MenuItem } from '@mui/material';
+import { useForm } from 'react-hook-form';
+import { FormattedMessage } from 'react-intl';
+import { CONTAINER_UNITS, IContainer } from '@/v5/store/containers/containers.types';
+import { NewFederation } from '@/v5/store/federations/federations.types';
+import { prepareNewFederation } from '@/v5/store/federations/federations.helpers';
+import { FederationsActionsDispatchers } from '@/v5/services/actionsDispatchers/federationsActions.dispatchers';
+import { SectionTitle } from '../federationSettingsForm/federationSettingsForm.styles';
+import { HalfWidth } from './createFederationForm.styles';
+import { EditFederation } from '../editFederationModal/editFederation';
+
+interface ICreateFederation {
+	open: boolean;
+	onClickClose: () => void;
+}
+
+const defaultValues = {
+	name: '',
+	unit: 'mm',
+};
+
+export const CreateFederationForm = ({ open, onClickClose }: ICreateFederation): JSX.Element => {
+	const {
+		handleSubmit,
+		control,
+		getValues,
+		reset,
+		formState: { errors, isValid, isSubmitSuccessful },
+	} = useForm<NewFederation>({
+		defaultValues,
+		mode: 'onChange',
+		resolver: yupResolver(FederationCreationSchema),
+	});
+	const { teamspace, project } = useParams() as { teamspace: string, project: string };
+	const [modalPhase, setModalPhase] = useState('settings');
+	const [includedContainers, setIncludedContainers] = useState([]);
+
+	const closeAndReset = (): void => {
+		onClickClose();
+		reset();
+		setModalPhase('settings');
+	};
+
+	useEffect(() => {
+		if (isSubmitSuccessful && (modalPhase === 'settings')) closeAndReset();
+	}, [open, reset]);
```

Comment:
I'm thinking, maybe this should be in the base dialog form? the reset bit? We need to check out how it impacts in other dialogs.

Reply:
Right now we have `FormDialog`s as a subcomponent of other custom react components. They are always rendered in the dom, but only show up when the property `open` is set to true. If we were to use a condition rendering inside the `formDialog`, it would fix the problem. Something like 
```ts
{open ? (
   <FormDialog>
      ...content
   </FormDialog>
): <></>}
```
This way the `FormDialog` also becomes agnostic about its conent

Reply:
Might there be some occasions where we don't want to reset the form when closed, only when submitted? Or am I misunderstanding what you're saying

Reply:
Lets think about that stuff in #3302